### PR TITLE
config: support lowercase proxy environment variables

### PR DESCRIPTION
**Pull request by sudomateo**
To my knowledge, there is no industry standard for the casing of proxy
environment variables. Applications can choose which case to honor (i.e.
`http_proxy` versus `HTTP_PROXY`).

Up to this point, Fluent Bit has only honored the uppercase proxy
environment variables `HTTP_PROXY` and `NO_PROXY`. This change adds
support for the lowercase variants of these environment variables
(`http_proxy` and `no_proxy`) while maintaining preference for the
uppercase variants when both uppercase and lowercase variants are
defined.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change
- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [x] Documentation required for this feature
  - [x] https://github.com/fluent/fluent-bit-docs/pull/658

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by edsiper** [comment]
cc: @erain

**Comment by patrick-stephens** [comment]
Unit tests are green, it's just waiting on the removed status we have now fixed.

**Review by nokute78 — APPROVED** [review]
LGTM.

**Review by erain — APPROVED** [review]
LGTM. Please also update the fluent-bit documentation for adding the lowercase envvar support.

**Comment by sudomateo** [comment]
Here is the documentation pull request: https://github.com/fluent/fluent-bit-docs/pull/658

**Comment by sudomateo** [comment]
Please let me know if there's anything else needed from me on this pull request.

I'm not sure who should review the documentation pull request but it is ready for review here: https://github.com/fluent/fluent-bit-docs/pull/658

**Comment by patrick-stephens** [comment]
The docs PR is just waiting on this merge so should be fine.

**Comment by edsiper** [comment]
thanks everyone!

**Pull request merged**
